app/controllers/api/base_controller/generic.rb

```diff
@@ -62,18 +62,14 @@ def delete_resource(type, id = nil, _data = nil)
       end
 
       def request_retire_resource(type, id, _data = nil)
-        if api_user_role_allows?('miq_request_approval')
```

Comment:
In talking to @gtanzillo about this issue it seems like this should actually be checking for the product feature that allows the user to retire the object.  There are different product features that would need to be checked based on the `type` value and the SUI now has its own set of product features which would need to be considered as well.  (See https://github.com/ManageIQ/manageiq-api/pull/501)

But my first question is if this check actually belongs here in the first place or should it be done elsewhere.  I do not see an other calls to `api_user_role_allows?` in this file and wonder if the role check is happening before we get here.

cc @abellotti

Reply:
I am just thinking (retiring is not my area):

and other question is whether method `request_retire_resource` is used just for services or also for other objects(Request, Orchestration Stack,,,,  ), maybe we should check "X_retire" permission for each type of object (X is. Service, Request, Orchestration Stack ) and check how retirement is done in UI classic (if it is present in UI and in case there is no API used yet)

Reply:
api_user_role_allows? is called in base_controller/parser.rb as we validate the action for the caller and decide on returning the 403's or not.

Reply:
The logic @abellotti is referring to is the call to `api_user_role_allows?` here https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/base_controller/parser.rb#L155.

So I believe we are doing the correct thing by removing this check.

I do however think we have a bug that was introduced with https://github.com/ManageIQ/manageiq-api/pull/501 where the service `retire` and `request_retire` product feature lookups are checking the wrong values.

Based on Product Features for
My Services:
https://github.com/ManageIQ/manageiq/blob/master/db/fixtures/miq_product_features.yml#L438-L441

and Service UI:
https://github.com/ManageIQ/manageiq/blob/master/db/fixtures/miq_product_features.yml#L6475-L6478

I would expect the following configuration in `config/api.yml`
```diff
--- a/config/api.yml
+++ b/config/api.yml
@@ -3234,12 +3234,12 @@
         - sui_services_edit
       - :name: retire
         :identifier:
-        - service_retire
+        - service_retire_now
         - sui_services_retire
       - :name: request_retire
         :identifier:
-        - service_retire
-        - sui_services_retire_date
+        - service_retire_now
+        - sui_services_retire
       - :name: set_ownership
         :identifier:
         - service_ownership
```

@gtanzillo @tinaafitz Does this look like the proper configuration change?

Reply:
@gmcculloug Yes, that looks like the proper configuration change.

Reply:
Looks correct to me @gmcculloug